### PR TITLE
Merge remaining sig_terms into terms

### PR DESCRIPTION
**Pull request by nik9000**
Merges the remaining implementation of `significant_terms` into `terms`
so that we can more easilly make them work properly without
`asMultiBucketAggregator` which *should* save memory and speed them up.

Relates #56487

### CODE REVIEW

**Comment by nik9000** [comment]
@elasticsearchmachine run elasticsearch-ci/default-distro

**Comment by nik9000** [comment]
run elasticsearch-ci/1

**Comment by nik9000** [comment]
run elasticsearch-ci/1

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java [hunk 1]
```diff
@@ -648,7 +647,6 @@ void updateBucket(OrdBucket spare, long globalOrd, long bucketOrd, long docCount
             spare.globalOrd = globalOrd;
             spare.bucketOrd = bucketOrd;
             spare.docCount = docCount;
-            otherDocCount += docCount;
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java:651]: This was causing inaccurate doc counts when `shard_min_doc_count` was more than 1.

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/NumericTermsAggregator.java [hunk 2]
```diff
@@ -164,17 +164,18 @@ public void collectDebugInfo(BiConsumer<String, Object> add) {
                 BucketOrdsEnum ordsEnum = bucketOrds.ordsEnum(owningBucketOrds[ordIdx]);
                 Supplier<B> emptyBucketBuilder = emptyBucketBuilder(owningBucketOrds[ordIdx]);
                 while (ordsEnum.next()) {
+                    long docCount = bucketDocCount(ordsEnum.ord());
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/NumericTermsAggregator.java:167]: I flip the `if` statement here and moved the spare initialization down to right before I need it to line up with the way the `map` version works.

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/StringTermsAggregator.java [hunk 3]
```diff
@@ -1,185 +0,0 @@
...
-import org.elasticsearch.search.internal.SearchContext;
-
-import java.io.IOException;
-import java.util.Arrays;
-import java.util.Map;
-
-/**
- * An aggregator of string values.
- */
-public class StringTermsAggregator extends AbstractStringTermsAggregator {
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/StringTermsAggregator.java:46]: Really just renamed this class to `MapStringTermsAggregator` but a lot changed.

**Review by not-napoleon — APPROVED** [review]
Looks good to me.  Much easier to see the relationship between these two aggs now.

**Pull request merged**